pkg/controller/controller_instance.go

```diff
@@ -716,6 +748,13 @@ func isServiceInstanceFailed(instance *v1beta1.ServiceInstance) bool {
 	return false
 }
 
+// isServiceInstanceProcessedAlready returns true if there is no further processing
+// needed for the instance based on ObservedGeneration
+func isServiceInstanceProcessedAlready(instance *v1beta1.ServiceInstance) bool {
+	return instance.Status.ObservedGeneration >= instance.Generation &&
+		(isServiceInstanceReady(instance) || isServiceInstanceFailed(instance))
```

Comment:
How would you feel about also adding a check for `!instance.Status.OrphanMitigationInProgress`? It doesn't change the behavior of the controller either way, because `isServiceInstanceProcessedAlready` is never called with orphan mitigation is in progress. However, I think it covers more completely the criteria that determine when a generation has been completely processed.